extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/VaultAuthManager.java

```diff
@@ -118,6 +138,50 @@ private VaultToken login(VaultAuthenticationType type) {
         return new VaultToken(auth.clientToken, auth.renewable, auth.leaseDurationSecs);
     }
 
+    private String getSecretId() {
+
+        Optional<String> secretIdOption = serverConfig.authentication.appRole.secretId;
+        if (secretIdOption.isPresent()) {
+            return secretIdOption.get();
+        }
+
+        return unwrapWrappingTokenOnce("secret id",
+                serverConfig.authentication.appRole.secretIdWrappingToken.get(), unwrap -> unwrap.data.secretId,
+                VaultAppRoleGenerateNewSecretID.class);
+    }
+
+    private <T> String unwrapWrappingTokenOnce(String type, String wrappingToken,
+            Function<T, String> f, Class<T> clazz) {
+
+        String wrappedValue = wrappedCache.get(wrappingToken);
```

Comment:
if the wrapped info is already in the cache, no need to go through semaphore acquisition
this will be faster, and that is the situation we will be in almost all the time